### PR TITLE
logseq: Add version 0.0.17

### LINKED ISSUE #5582: [Request]  Add logseq
A privacy-first, open-source(frontend-only) platform for knowledge sharing and management.
https://github.com/logseq/logseq

### PR DESCRIPTION
**Pull request by HueLiu**
- Closes #5582

### CODE REVIEW

File: bucket/logseq.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{
+    "version": "0.0.16",
+    "description": "Logseq is a privacy-first, open-source(frontend now, backend later) platform for knowledge sharing and management.",
+    "homepage": "https://logseq.com/",
+    "license": {
+        "identifier": "GNU Affero General Public License v3.0",
```
Comment (Ash258) [inline bucket/logseq.json:6]: Invalid SPDX identifier

File: bucket/logseq.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        "url": "https://github.com/logseq/logseq/blob/master/LICENSE.md"
+    },
+    "notes": "We don't persist your Logseq data, they are still storaged in \"%APPDATA%\\Logseq\".",
```
Comment (Ash258) [inline bucket/logseq.json:9]: Not needed message

```suggestion
```

File: bucket/logseq.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
+{
+    "version": "0.0.16",
+    "description": "Logseq is a privacy-first, open-source(frontend now, backend later) platform for knowledge sharing and management.",
```
Comment (Ash258) [inline bucket/logseq.json:3]: Mentioning application name and that it is open source, is redundant

File: bucket/logseq.json [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    "hash": "bf1dc4e7b8219724926b77fd8b4cfdb3a1483ac0a3120020d7297096881e88be",
+    "installer": {
+        "script": [
+            "$nupkg = (Get-ChildItem \"$dir\" 'logseq-*-full.nupkg' | Select-Object -First 1).FullName",
+            "Expand-7ZipArchive \"$nupkg\" \"$dir\" -Removal",
+            "$dotnet = (Get-ChildItem \"$dir\\lib\" 'net*' | Select-Object -First 1).FullName",
+            "Move-Item \"$dotnet\\*\" \"$dir\""
+        ]
+    },
+    "post_install": "Remove-Item \"$dir\\lib\",\"$dir\\_rels\" -Recurse -Force",
```
Comment (Ash258) [inline bucket/logseq.json:20]: Nonse to have special install block for removing files

File: bucket/logseq.json [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    "description": "Logseq is a privacy-first, open-source(frontend now, backend later) platform for knowledge sharing and management.",
+    "homepage": "https://logseq.com/",
+    "license": {
+        "identifier": "GNU Affero General Public License v3.0",
+        "url": "https://github.com/logseq/logseq/blob/master/LICENSE.md"
+    },
+    "notes": "We don't persist your Logseq data, they are still storaged in \"%APPDATA%\\Logseq\".",
+    "url": "https://github.com/logseq/logseq/releases/download/0.0.16/logseq-win-x64-0.0.16.exe#/dl.7z",
+    "hash": "bf1dc4e7b8219724926b77fd8b4cfdb3a1483ac0a3120020d7297096881e88be",
+    "installer": {
```
Comment (Ash258) [inline bucket/logseq.json:12]: Convert to pre_install

File: bucket/logseq.json [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+        "script": [
+            "$nupkg = (Get-ChildItem \"$dir\" 'logseq-*-full.nupkg' | Select-Object -First 1).FullName",
+            "Expand-7ZipArchive \"$nupkg\" \"$dir\" -Removal",
```
Comment (Ash258) [inline bucket/logseq.json:15]: Expand archive accept wildcard

File: bucket/logseq.json [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+            "$dotnet = (Get-ChildItem \"$dir\\lib\" 'net*' | Select-Object -First 1).FullName",
```
Comment (Ash258) [inline bucket/logseq.json:16]: Variable is not needed

**Review by Ash258 — CHANGES_REQUESTED** [review]

**Comment by HueLiu** [comment]
Hi,

Since I am new to scoop, there are many mistakes. Thank you very much for your review and suggestions.

I've updated some of the code, I wasn't sure if it was appropriate, thanks!

File: bucket/logseq.json [hunk 8]
```diff
@@ -0,0 +1,28 @@
+{
+    "version": "0.0.16",
+    "description": "A privacy-first platform for knowledge sharing and management.",
```
Comment (Ash258) [inline bucket/logseq.json:3]:
```suggestion
    "description": "A privacy-first platform for knowledge sharing and management",
```

File: bucket/logseq.json [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+    "homepage": "https://logseq.com/",
```
Comment (Ash258) [inline bucket/logseq.json:4]:
```suggestion
    "homepage": "https://logseq.com",
```

File: bucket/logseq.json [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+    "license": {
+        "identifier": "AGPL-3.0-only",
+        "url": "https://github.com/logseq/logseq/blob/master/LICENSE.md"
+    },
```
Comment (Ash258) [inline bucket/logseq.json:8]:
```suggestion
    "license": "AGPL-3.0-only",
```

File: bucket/logseq.json [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+        [
+            "Logseq.exe",
+            "Logseq"
+        ]
+    ],
+    "checkver": {
+        "github": "https://github.com/logseq/logseq"
+    },
+    "autoupdate": {
+        "url": "https://github.com/logseq/logseq/releases/download/$version/logseq-win-x64-$version.exe#/dl.7z"
```
Comment (Ash258) [inline bucket/logseq.json:26]:
```suggestion
        "architecture": {
            "64bit": {
                "url": "https://github.com/logseq/logseq/releases/download/$version/logseq-win-x64-$version.exe#/dl.7z"
            }
        }
```

File: bucket/logseq.json [hunk 12]
```diff
@@ -0,0 +1,28 @@
+{
+    "version": "0.0.16",
```
Comment (Ash258) [inline bucket/logseq.json:2]:
```suggestion
    "version": "0.0.17",
```

File: bucket/logseq.json [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+    "description": "A privacy-first platform for knowledge sharing and management.",
+    "homepage": "https://logseq.com/",
+    "license": {
+        "identifier": "AGPL-3.0-only",
+        "url": "https://github.com/logseq/logseq/blob/master/LICENSE.md"
+    },
+    "url": "https://github.com/logseq/logseq/releases/download/0.0.16/logseq-win-x64-0.0.16.exe#/dl.7z",
+    "hash": "bf1dc4e7b8219724926b77fd8b4cfdb3a1483ac0a3120020d7297096881e88be",
```
Comment (Ash258) [inline bucket/logseq.json:10]:
```suggestion
    "architecture": {
        "64bit": {
            "url": "https://github.com/logseq/logseq/releases/download/0.0.17/logseq-win-x64-0.0.17.exe#/dl.7z",
            "hash": "12d342d26d46ed4ebe97891a2798e482ab0161d79def27395cab654062966ee0"
        }
    },
```

File: bucket/logseq.json [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+    "pre_install": [
+            "Expand-7ZipArchive \"$dir\\logseq-*-full.nupkg\" \"$dir\" -Removal",
+            "Move-Item \"$dir\\lib\\net45\\*\" \"$dir\"",
+            "Remove-Item \"$dir\\*.gif\", \"$dir\\_rels\" , \"$dir\\lib\", \"$dir\\package\", \"$dir\\*.xml\"  -Recurse"
```
Comment (Ash258) [inline bucket/logseq.json:14]:
```suggestion
        "Expand-7ZipArchive \"$dir\\logseq-*-full.nupkg\" -ExtractDir 'lib\\net45' -Removal",
        "Remove-Item \"$dir\\lib\", \"$dir\\Update*\" -Recurse"
```

**Review by Ash258 — CHANGES_REQUESTED** [review]

**Review by Ash258 — CHANGES_REQUESTED** [review]
https://github.com/lukesampson/scoop-extras/pull/5951#discussion_r606054476

**Review by Ash258 — APPROVED** [review]

**Pull request merged**